### DISCUSSION TITLE
Spacing between NavigationBar elements on Android

### DISCUSSION THREAD

**Discussion by astanb** [grahammendick/navigation#573] [Q&A] [answered]
I'm having a real hard time getting the nav bar to look like I want. It seems the `navigationImage` is set at a different inset to `RightBar > BarButton` meaning the spacing on the outside is different. Plus the container that `navigationImage` is in seems to have a preset width that cannot be altered, meaning getting my branding to sit in the middle of the two is also impossible to achieve without some hacks.

Please see the attached image for what I am trying to achieve (you should assume the gaps are the same i.e. the positioning is symmetrical).

![image](https://user-images.githubusercontent.com/809537/151596831-d2bd309c-8c1b-4c6c-87f7-ee060b3c9ba7.png)

Am I missing something here?

**Comment by grahammendick** [comment, marked as answer]
Hi, nice to hear from you. The `NavigationBar` renders to the native `Toolbar` on Android so I think the spacing matches the [material design specs](https://material.io/components/app-bars-top#specs). I'm not manually setting any spacing. Could you compare it to some native Android apps on your phone, like WhatsApp or Twitter, and see if there's a difference in spacing?

![Screenshot 2022-01-28 at 18 13 19](https://user-images.githubusercontent.com/1761227/151599728-3d6f2a7c-15bf-4e27-959f-72da300c47be.png)

I'm guessing your branding is a `TitleBar`. I use something like the following to [center the title on the zoom sample](https://github.com/grahammendick/navigation/blob/df28e810e070f2a61af9918c0b43366b292c7524/NavigationReactNative/sample/zoom/Detail.js#L13)
```jsx
<NavigationBar>
  <TitleBar style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}>
     <Branding style={{width: 200}}>
  </TitleBar>
</NavigationBar>
```

**Reply by astanb** [reply]
Hi Graham, thanks for the detailed reply. I was able to resolve it using the Material measurements and adding a marginLeft of 16 to the `TitleBar`. Not my favourite hack but it works!
